packages/evm/core/src/state_hash.rs

```diff
@@ -0,0 +1,32 @@
+use rayon::slice::ParallelSliceMut;
+use revm::primitives::{keccak256, B256};
+
+use crate::{state_changes::StateChangeset, state_commit::StateCommit};
+
+pub fn calculate(current_hash: B256, state: &StateCommit) -> Result<B256, crate::db::Error> {
+    let commit_hash = bincode::serialize(&prepare(state))?;
```

Comment:
Commit hash is actually result of next line. At this point we are dealing with serialized non hashed object. 

```suggestion
    let serialized = bincode::serialize(&prepare(state))?;
```